File: backend/app/schemas/business_health.py
"""Executive business health dashboard schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class BusinessHealthMetrics(BaseModel):
    revenue_today_inr: str
    revenue_month_inr: str
    revenue_growth_pct: float | None = None
    orders_today: int
    orders_month: int
    order_growth_pct: float | None = None
    average_order_value_inr: str
    active_customers: int
    new_customers: int
    returning_customers: int
    active_laundries: int
    top_laundry_name: str | None = None
    top_laundry_revenue_inr: str | None = None
    lowest_laundry_name: str | None = None
    lowest_laundry_revenue_inr: str | None = None
    total_customers: int
    total_laundries: int


class BusinessHealthChartPoint(BaseModel):
    date: str
    value: float
    label: str | None = None


class BusinessHealthMonthPoint(BaseModel):
    month: str
    count: int


class BusinessHealthCharts(BaseModel):
    revenue_trend: list[BusinessHealthChartPoint] = Field(default_factory=list)
    orders_trend: list[BusinessHealthChartPoint] = Field(default_factory=list)
    customer_growth: list[BusinessHealthMonthPoint] = Field(default_factory=list)
    laundry_growth: list[BusinessHealthMonthPoint] = Field(default_factory=list)
    commission_trend: list[BusinessHealthChartPoint] = Field(default_factory=list)


class BusinessHealthOperational(BaseModel):
    open_disputes: int
    pending_refunds: int
    pending_settlements: int
    failed_deliveries: int
    delayed_orders: int
    delayed_settlements: int


class BusinessHealthGrowth(BaseModel):
    customer_growth_pct: float | None = None
    laundry_growth_pct: float | None = None
    order_growth_pct: float | None = None
    revenue_growth_pct: float | None = None
    new_customers_month: int
    new_laundries_month: int


class BusinessHealthAlert(BaseModel):
    id: str
    severity: str
    title: str
    description: str
    metric_value: str
    href: str


class BusinessHealthTrendPoint(BaseModel):
    date: str
    revenue_inr: str
    orders: int


class BusinessHealthDashboardResponse(BaseModel):
    metrics: BusinessHealthMetrics
    operational: BusinessHealthOperational
    growth: BusinessHealthGrowth
    alerts: list[BusinessHealthAlert] = Field(default_factory=list)
    trend: list[BusinessHealthTrendPoint] = Field(default_factory=list)
    charts: BusinessHealthCharts = Field(default_factory=BusinessHealthCharts)
    generated_at: str

File: frontend/services/business-health.ts
import { api, type ApiEnvelope } from '@/lib/api';

export interface BusinessHealthMetrics {
  revenue_today_inr: string;
  revenue_month_inr: string;
  revenue_growth_pct: number | null;
  orders_today: number;
  orders_month: number;
  order_growth_pct: number | null;
  average_order_value_inr: string;
  active_customers: number;
  new_customers: number;
  returning_customers: number;
  active_laundries: number;
  top_laundry_name: string | null;
  top_laundry_revenue_inr: string | null;
  lowest_laundry_name: string | null;
  lowest_laundry_revenue_inr: string | null;
  total_customers: number;
  total_laundries: number;
}

export interface BusinessHealthOperational {
  open_disputes: number;
  pending_refunds: number;
  pending_settlements: number;
  failed_deliveries: number;
  delayed_orders: number;
  delayed_settlements: number;
}

export interface BusinessHealthGrowth {
  customer_growth_pct: number | null;
  laundry_growth_pct: number | null;
  order_growth_pct: number | null;
  revenue_growth_pct: number | null;
  new_customers_month: number;
  new_laundries_month: number;
}

export interface BusinessHealthAlert {
  id: string;
  severity: 'warning' | 'critical' | string;
  title: string;
  description: string;
  metric_value: string;
  href: string;
}

export interface BusinessHealthTrendPoint {
  date: string;
  revenue_inr: string;
  orders: number;
}

export interface BusinessHealthChartPoint {
  date: string;
  value: number;
  label: string | null;
}

export interface BusinessHealthMonthPoint {
  month: string;
  count: number;
}

export interface BusinessHealthCharts {
  revenue_trend: BusinessHealthChartPoint[];
  orders_trend: BusinessHealthChartPoint[];
  customer_growth: BusinessHealthMonthPoint[];
  laundry_growth: BusinessHealthMonthPoint[];
  commission_trend: BusinessHealthChartPoint[];
}

export interface BusinessHealthDashboard {
  metrics: BusinessHealthMetrics;
  operational: BusinessHealthOperational;
  growth: BusinessHealthGrowth;
  alerts: BusinessHealthAlert[];
  trend: BusinessHealthTrendPoint[];
  charts: BusinessHealthCharts;
  generated_at: string;
}

export async function getBusinessHealthDashboard(): Promise<BusinessHealthDashboard> {
  const { data } = await api.get<ApiEnvelope<BusinessHealthDashboard>>('/admin/business-health');
  return data.data;
}

export function formatGrowthPct(pct: number | null | undefined): string {
  if (pct == null) return '—';
  return `${pct >= 0 ? '+' : ''}${pct}%`;
}
